deepchem/metrics/__init__.py

```diff
@@ -314,132 +609,131 @@ def __init__(self,
       ]:
         mode = "regression"
       else:
-        raise ValueError("Must specify mode for new metric.")
-    assert mode in ["classification", "regression"]
-    if self.metric.__name__ in [
-        "accuracy_score", "balanced_accuracy_score", "recall_score",
-        "matthews_corrcoef", "precision_score", "f1_score"
-    ] and threshold is None:
-      self.threshold = 0.5
-    self.mode = mode
-    # The convention used is that the first task is the metric.
-    # TODO(rbharath, joegomes): This doesn't seem like it should be hard-coded as
-    # an option in the Metric class. Instead, this should be possible to move into
-    # user-space as a custom task_averager function.
-    self.compute_energy_metric = compute_energy_metric
+        logger.info(
+            "Could not detect mode of classifier. Check your results carefully."
+        )
+      self.mode = mode
 
   def compute_metric(self,
                      y_true,
                      y_pred,
                      w=None,
                      n_classes=2,
-                     filter_nans=True,
-                     per_task_metrics=False):
+                     filter_nans=False,
+                     per_task_metrics=False,
+                     use_sample_weights=False,
+                     threshold=None):
     """Compute a performance metric for each task.
 
     Parameters
     ----------
     y_true: np.ndarray
-      An np.ndarray containing true values for each task.
+      An np.ndarray containing true values for each task. Must be of
+      shape `(N, n_tasks, n_classes)` if a classification metric, else
+      must be of shape `(N, n_tasks)` if a regression metric.
     y_pred: np.ndarray
-      An np.ndarray containing predicted values for each task.
+      An np.ndarray containing predicted values for each task. Must be
+      of shape `(N, n_tasks, n_classes)` if a classification metric,
+      else must be of shape `(N, n_tasks)` if a regression metric.
     w: np.ndarray, optional
-      An np.ndarray containing weights for each datapoint.
+      An np.ndarray containing weights for each datapoint. If
+      specified,  must be of shape `(N, n_tasks)`.
```

Comment:
Except they don't actually have to have those shapes, because the methods you wrote make heroic efforts to figure out how to interpret things of all sorts of different shapes!

Reply:
Ah good point! Will fix the docs for this